mmgen/core/hooks/ema_hook.py

```diff
@@ -61,14 +66,49 @@ def __init__(self,
         assert hasattr(
             self, interp_mode
         ), f'Currently, we do not support {self.interp_mode} for EMA.'
-        self.interp_func = partial(
-            getattr(self, interp_mode), **self.interp_cfg)
+        self.interp_func = getattr(self, interp_mode)
+
+        self.momentum_cfg = dict() if momentum_cfg is None else deepcopy(
+            momentum_cfg)
+        self.momentum_policy = momentum_policy
+        if momentum_policy != 'fixed':
+            assert hasattr(
+                self, momentum_policy
+            ), f'Currently, we do not support {self.momentum_policy} for EMA.'
+            self.momentum_updater = getattr(self, momentum_policy)
 
     @staticmethod
     def lerp(a, b, momentum=0.999, momentum_nontrainable=0., trainable=True):
         m = momentum if trainable else momentum_nontrainable
         return a + (b - a) * m
 
+    @staticmethod
+    def rampup(runner, ema_kimg=10, ema_rampup=0.05, batch_size=4, eps=1e-8):
```

Comment:
Maybe we should change the naming template of momentum update policy as `{POLICY}_momentum_updater` or `get_{POLICY}_momentum`.
 Because we had already used the name of interpolating policy (e.g. `lerp`) as the function name. Using momentum update policy as function name may lead to confusion.

Reply:
may be we can register interp_func and momentum_updater in this class? Keep these functions in two separate function dictionaries.